server/api/tag/__tests__/tag.spec.js

```diff
@@ -4,37 +4,135 @@ import { server, appReady } from '../../../server'
 })
 
 test.after.always(async (t) => {
+  await Tag.deleteMany()
   await t.context.memMongo.stop()
 })
 
-test.beforeEach('connect and add tag entries', async (t) => {
-  t.context.tags = await Tag.create({ tags: tags }).catch((err) => console.error('Unable to create tags', err))
+/* -- Anon person */
+test('Tag API - anon - create', async t => {
+  const response = await request(server)
+    .post('/api/tags')
+    .send({ tags: ['a', 'b', 'c'] })
+
+  t.is(response.statusCode, 403)
 })
 
-test.afterEach.always(async () => {
-  await Tag.deleteMany()
+test('Tag API - anon - read', async t => {
+  const res = await request(server)
+    .get('/api/tags')
+    .set('Accept', 'application/json')
+    .expect(403)
+  t.is(res.status, 403)
 })
+test('Tag API - anon - update', async t => {
+  const tag = t.context.tags[0]
 
-test.serial('verify fixture database has tags', async t => {
-  const p = await Tag.find()
-  t.is(1, p.length)
-  t.truthy(p[0].tags.includes('robots'))
+  const response = await request(server)
+    .put(`/api/tags/${tag._id}`)
+    .send({ tags: ['a', 'b', 'c'] })
+
+  t.is(response.statusCode, 403)
 })
 
-test.serial('Should correctly give count of all tags', async t => {
+test('Tag API - anon - delete', async t => {
+  const tag = t.context.tags[0]
+
+  const response = await request(server)
+    .delete(`/api/tags/${tag._id}`)
+
+  t.is(response.statusCode, 403)
+})
+
+/* -- Signed in person */
+
+test('Tag API - signed - create', async t => {
+  const response = await request(server)
+    .post('/api/tags')
+    .set('Cookie', [`idToken=${jwtDataDali.idToken}`])
+    .send({ tags: ['a', 'b', 'c'] })
+
+  t.is(response.statusCode, 403)
+})
+
+test('Tag API - signed - read', async t => {
   const res = await request(server)
     .get('/api/tags')
     .set('Accept', 'application/json')
+    .set('Cookie', [`idToken=${jwtDataDali.idToken}`])
     .expect(200)
     .expect('Content-Type', /json/)
+
   const got = res.body
   t.is(tags.length, got.length)
 })
+
+test('Tag API - signed - update', async t => {
+  const tag = t.context.tags[0]
+
+  const response = await request(server)
+    .put(`/api/tags/${tag._id}`)
+    .set('Cookie', [`idToken=${jwtDataDali.idToken}`])
+    .send({ tags: ['a', 'b', 'c'] })
+
+  t.is(response.statusCode, 403)
+})
+
+test('Tag API - signed - delete', async t => {
+  const tag = t.context.tags[0]
+
+  const response = await request(server)
+    .delete(`/api/tags/${tag._id}`)
+    .set('Cookie', [`idToken=${jwtDataDali.idToken}`])
+
+  t.is(response.statusCode, 403)
+})
+
+/* -- Admin person */
+
+test('Tag API - admin - CRUD', async t => {
+  let response = await request(server)
+    .post('/api/tags')
+    .set('Cookie', [`idToken=${jwtData.idToken}`])
+    .send({ tags: ['a', 'b', 'c'] })
+
+  t.is(response.statusCode, 200)
+  const newtags = response.body
+  const id = newtags._id
+
+  // get by id - non admin check
+  response = await request(server)
+    .put(`/api/tags/${id}`)
```

Comment:
Similar question to the create one here - is the client ever going to be updating the list of tags (and actually need this access)? It seems like there is a lot of room for error/bugs here - given there is only one record for all tags I could see it being easy to accidentally overwrite the entire list and lose data

Reply:
Just noticed that based on the comments that this possibly should have been a `.get` call?

Reply:
Admins will get a tag list management UI sometime.  As tags are a pick list and not referenced in the tag arrays of entities, updating the list does not break references where other words have been used. 
Aim is to populate a large list of standard terms. 

good spot - will update.